src/main/kotlin/org/jetbrains/research/testspark/appstarter/TestSparkStarter.kt

```diff
@@ -0,0 +1,219 @@
+                    // First, get CUT Module
+                    val cutModule =
+                        ProjectFileIndex.getInstance(project)
+                            .getModuleForFile(targetPsiClass.containingFile.virtualFile)!!
+                    // Then, instantiate the project context
+                    val projectContext = ProjectContext(
+                        classPath,
+                        output,
+                        targetPsiClass,
+                        targetPsiClass.qualifiedName,
+                        cutModule
+                    )
+                    // Prepare the test generation data
+                    val testGenerationData = TestGenerationData(
+                        resultPath = output,
+                        testResultName = "HeadlessGeneratedTests"
+                    )
+                    println("[TestSpark Starter] Indexing is done")
+
+                    // get package name
+                    val packageList = targetPsiClass.qualifiedName.toString().split(".").toMutableList()
+                    packageList.removeLast()
+                    val packageName = packageList.joinToString(".")
+
+                    // Create a process Manager
+                    val llmProcessManager = Llm()
+                        .getLLMProcessManager(
+                            project,
+                            psiFile,
+                            targetPsiClass.textRange.startOffset,
+                            ""
+                        )
+
+                    println("[TestSpark Starter] Starting the test generation process")
+                    // Start test generation
+                    val indicator = HeadlessProgressIndicator()
+                    val errorMonitor = DefaultErrorMonitor()
+                    val uiContext = llmProcessManager.runTestGenerator(
+                        indicator,
+                        FragmentToTestData(CodeType.CLASS),
+                        packageName,
+                        projectContext,
+                        testGenerationData,
+                        errorMonitor
+                    )
+
+                    // Check test Generation Output
+                    if (uiContext != null) {
+                        println("[TestSpark Starter] Test generation completed successfully")
+                        // Run test file
+                        runTests(project, output, packageList, classPath, projectContext)
+                    } else {
+                        println("[TestSpark Starter] Test generation failed")
+                    }
+
+                    ProjectManager.getInstance().closeAndDispose(project)
+
+                    println("[TestSpark Starter] Exiting the headless mode")
+                    exitProcess(0)
+                }
+            }
+        }
+    }
+
+    private fun runTests(project: Project, out: String, packageList: MutableList<String>, classPath: String, projectContext: ProjectContext) {
```

Comment:
It is better to name the function differently, as it is not clear what does: `runTestsAndCollectCoverage`/`runTestsWithCoverageCollection`; something like that.